File: app/src/main/java/com/revosleap/bxplayer/AppUtils/BxPlayback/BxPlayerService.java
package com.revosleap.bxplayer.AppUtils.BxPlayback;

import android.app.Service;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.media.AudioManager;
import android.os.Binder;
import android.os.IBinder;
import android.widget.Toast;

public class BxPlayerService extends Service {
    private final IBinder mIBinder = new BXBinder();

    private MediaPlayerHolder mMediaPlayerHolder;

    private BXNotificationManager mMusicNotificationManager;

    private boolean sRestoredFromPause = false;

    public final boolean isRestoredFromPause() {
        return sRestoredFromPause;
    }

    public void setRestoredFromPause(boolean restore) {
        sRestoredFromPause = restore;
    }

    public final MediaPlayerHolder getMediaPlayerHolder() {
        return mMediaPlayerHolder;
    }

    public BXNotificationManager getMusicNotificationManager() {
        return mMusicNotificationManager;
    }


    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        mMediaPlayerHolder.registerNoisyReceiver();
        return START_NOT_STICKY;
    }
    @Override
    public void onDestroy() {
        mMediaPlayerHolder.registerNotificationActionsReceiver(false);
        mMusicNotificationManager = null;
        mMediaPlayerHolder.release();
        super.onDestroy();
    }

    @Override
    public IBinder onBind(Intent intent) {
        if (mMediaPlayerHolder == null) {
            mMediaPlayerHolder = new MediaPlayerHolder(this);
            mMusicNotificationManager = new BXNotificationManager(this);
            mMediaPlayerHolder.registerNotificationActionsReceiver(true);
        }
        return mIBinder;
    }
    public class BXBinder extends Binder{
        public BxPlayerService getInstance(){
            return BxPlayerService.this;
        }
    }

}

File: app/src/main/java/com/revosleap/bxplayer/utils/playback/BxPlayerService.kt
package com.revosleap.bxplayer.utils.playback

import android.app.Service
import android.content.Intent
import android.os.Binder
import android.os.IBinder

class BxPlayerService : Service() {
    private val mIBinder = BXBinder()

    var mediaPlayerHolder: MediaPlayerHolder? = null
        private set

    var musicNotificationManager: BXNotificationManager? = null
        private set

    var isRestoredFromPause = false


    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        mediaPlayerHolder!!.registerNoisyReceiver()
        return Service.START_NOT_STICKY
    }

    override fun onDestroy() {
        mediaPlayerHolder!!.registerNotificationActionsReceiver(false)
        musicNotificationManager = null
        mediaPlayerHolder!!.release()
        super.onDestroy()
    }

    override fun onBind(intent: Intent): IBinder? {
        if (mediaPlayerHolder == null) {
//            mediaPlayerHolder = MediaPlayerHolder(this)
//            musicNotificationManager = BXNotificationManager(this)
//            mediaPlayerHolder!!.registerNotificationActionsReceiver(true)
        }
        return mIBinder
    }

    inner class BXBinder : Binder() {
        val instance: BxPlayerService
            get() = this@BxPlayerService
    }

}
